proposed/building-arch-specific-applications.md

```diff
@@ -0,0 +1,806 @@
+# Building architecture-specific applications
+
+The first set of .NET Core releases, up to and including .NET Core 2.2, focused on the needs of web applications built with ASP.NET Core. Web applications don't directly depend on native libraries or OS-specific APIs as a general rule. As a result, we didn't focus on the usability of producing architecture-specific applications or that depend on native dependencies as much as we need to looking forward.
+
+Client applications differ in this respect, often relying on native libraries and APIs and typically need to integrate with OS-provided experiences (like notifications). With the addition of client scenarios in .NET Core 3.0, we need to reassess the usability and capability of building applications and libraries that that are intended for only a single architecture.
+
+Our goal is that `dotnet build` should produce useful and intuitive artifacts for building both client and server applications. Where a non-default configuration is desired, it should be easy and intutive to select that option.
+
+The document proposes the following:
+
+* .NET Core applications should be architecture-neutral by default.
+* Client applications should be architecture-specific by default, controlled via templates.
+* Only architecture-specific (including self-contained) applications should have EXEs.
+* The default for `--self-contained` is bad for usability. We should fix that for 3.0+ projects.
+
+Note: The term *architecture* is used throughout this document. It is intended to mean the unique combination of an operating system and CPU, like Windows X64, or Linux ARM64. It is equivalent to runtime identifier (RID).
+
+Note: There are separate challenges on building libraries that have native dependencies. That issue is not covered in this document, but is a problem that will be discussed in a following document.
+
+## Current Behavior
+
+The default `dotnet build` behavior in .NET Core 3.0 Preview 4 is the following:
```

Comment:
Can you clarify whether this is also the .NET Core 2.2 behavior in the doc. IOW, is this a preview over preview or release over release change.

Reply:
It's mostly the same as 2.2. I clarified in an update to the doc.